src/parser/parser.cpp

```diff
@@ -154,56 +154,58 @@ void Parser::ParseQuery(const string &query) {
 				transformer.TransformParseTree(parser.parse_tree, statements);
 			} else {
 				parsing_failed = true;
+				parser_error = QueryErrorContext::Format(query, parser.error_message, parser.error_location - 1);
 			}
 		}
-		// if DuckDB fails to parse the entire sql string, break the string down into individual statements
-		// so that parser extensions can help parse some statements
-		if (parsing_failed) {
+		// If DuckDB fails to parse the entire sql string, break the string down into individual statements
+		// using ; as the delimiter so that parser extensions can parse the statement
+		if (parsing_failed && options.extensions) {
```

Comment:
It a little hard to tell with nesting (be nice to split up for readability), but think we might be missing the `else` block corresponding with this `if` that throws the original error. If parsing failed but their is `no options.extensions` [will need to split up this if.

I suggest (to make it straight forward to check we have the right logic)
```
if(parsing_succeed) {
    return
} else if(!options.extensions) {
    throw parer_error
} else {
    [split logic]
}
```

Reply:
Totally agree, a big `if` is def bad. I refactored the logic here: https://github.com/duckdb/duckdb/pull/7868/commits/818e9ccb61ba9386444e744b56d78d1c3de2081f Could you take another look please?

Reply:
Looks good